### PR TITLE
chore: clean up test workflow

### PR DESCRIPTION
**Pull request by jonkoops**
**What kind of change does this PR introduce?**
- [ ] Bug Fix
- [ ] Feature
- [ ] Refactoring
- [ ] Style
- [ ] Build
- [x] Chore
- [ ] Documentation
- [ ] CI

**Did you add tests for your changes?**
- [ ] Yes, my code is well tested
- [x] Not relevant

**If relevant, did you update the documentation?**
- [ ] Yes, I've updated the documentation
- [x] Not relevant

**Summary**
Cleans up the test workflow to use all the latest versions of various actions.

**Does this PR introduce a breaking change?**
No.

**Other information**
None.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -5,41 +5,47 @@ on:
...
+          flag-name: run-${{ join(matrix.*, '-') }}
           parallel: true
 
   coverage:
+    name: Finish coverage
     needs: test
     runs-on: ubuntu-latest
     steps:
-      - uses: coverallsapp/github-action@v1.0.1
+      - name: Finish coverage
```
Comment (rolandjitsu) [inline .github/workflows/test.yml:48]:
```suggestion
      - name: Publish coverage
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -5,41 +5,47 @@ on:
...
+
+      - name: Collect coverage
+        uses: coverallsapp/github-action@v2
         with:
-          github-token: ${{ secrets.GITHUB_TOKEN }}
+          flag-name: run-${{ join(matrix.*, '-') }}
           parallel: true
 
   coverage:
+    name: Finish coverage
```
Comment (rolandjitsu) [inline .github/workflows/test.yml:44]:
```suggestion
    name: Publish coverage
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -5,41 +5,47 @@ on:
       - master
       - "[0-9]+.[0-9]+.x"
   pull_request:
-    types: [opened, synchronize]
```
Comment (rolandjitsu) [inline .github/workflows/test.yml:8]: Why remove this? We don't want to run tests for every single branch unless it's beta/master/maintenance and for PRs.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -5,41 +5,47 @@ on:
...
 
 jobs:
   test:
+    name: Run tests on Node.js ${{ matrix.node-version }}
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        node_version: [14, 16, 18, 20, 22]
+        node-version: [14, 16, 18, 20, 22]
```
Comment (rolandjitsu) [inline .github/workflows/test.yml:16]: Why the var name change?

**Review by rolandjitsu — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 3 above)
Comment (jonkoops) [reply .github/workflows/test.yml:8]: The default types are `opened`, `synchronize` and `reopened`, so falling back to the default here will result in pretty much exactly the desired behaviour.

File: .github/workflows/test.yml (hunk 4 above)
Comment (jonkoops) [reply .github/workflows/test.yml:16]: Convention, all GitHub Actions use this type of casing.

File: .github/workflows/test.yml (hunk 1 above)
Comment (jonkoops) [reply .github/workflows/test.yml]: Done

File: .github/workflows/test.yml (hunk 2 above)
Comment (jonkoops) [reply .github/workflows/test.yml]: Done

**Comment by jonkoops** [comment]
@rolandjitsu would you like me to make any further changes, or can we get this merged?

File: .github/workflows/test.yml (hunk 3 above)
Comment (rolandjitsu) [reply .github/workflows/test.yml:8]: I just find it odd to leave the `pull_request` key without any props 😆  In the spirit of being explicit as you were with the other changes, I'd prefer to keep this one as well.

**Review by rolandjitsu — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 3 above)
Comment (jonkoops) [reply .github/workflows/test.yml:8]: It's pretty common to do this in GitHub Actions, for example many of the examples in [the documentation](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows) only specify a trigger without a type. But if you feel strongly about it I will change it.

**Comment by jonkoops** [comment]
@rolandjitsu added the suggested changes, let me know what you think.

**Comment by jonkoops** [comment]
I went ahead and marked all the other open PRs as drafts as I want to merge this one one in first and rebase the rest on top. @rolandjitsu if you have no further objections I would like to merge this in.

**Review by rolandjitsu — APPROVED** [review]

**Comment by jonkoops** [comment]
Thanks! Let me see if I can get the others in a mergable state.

**Pull request merged**